src/Collections/Arrays/BinarySearch.dfy

```diff
@@ -0,0 +1,41 @@
+// RUN: %dafny /compile:0 "%s"
+
+/*******************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT 
+*******************************************************************************/
+
+include "../../StandardLibrary.dfy"
+include "../../UInt.dfy"
+
+module BinarySearch {
+  export
+    provides StandardLibrary, UInt
+
+  import StandardLibrary
+  import opened UInt = StandardLibrary.UInt
+  import opened Wrappers = StandardLibrary.Wrappers
+
+method BinarySearch<T>(a: array<T>, key: T, compare: (T, T) -> bool) returns (r:Option<nat>)
```

Comment:
It'd be nice to have this indented two spaces to be consistent with the imports above it.